pkg/reconciler/apibinding/apibinding_indexes_test.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apibinding
+
+import (
+	"reflect"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clusters"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+)
+
+func TestIndexAPIBindingByWorkspaceExport(t *testing.T) {
+	tests := map[string]struct {
+		obj     interface{}
+		want    []string
+		wantErr bool
+	}{
+		"not an APIBinding": {
+			obj:     "not an APIBinding",
+			want:    []string{},
+			wantErr: true,
+		},
+		"no workspace reference": {
+			obj:     &apisv1alpha1.APIBinding{},
+			want:    []string{},
+			wantErr: false,
+		},
+		"has a workspace reference": {
+			obj: &apisv1alpha1.APIBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					ClusterName: "root:default",
+					Name:        "foo",
+				},
+				Spec: apisv1alpha1.APIBindingSpec{
+					Reference: apisv1alpha1.ExportReference{
+						Workspace: &apisv1alpha1.WorkspaceExportReference{
+							WorkspaceName: "workspace1",
```

Comment:
(No action required) Are apibindings only ever referring to workspaces that are a peer to its own?

Reply:
see API docs. It can be an absolute reference

Reply:
In that case, maybe add a test case to validate handling an absolute reference? It's not clear to me how the method under test would correctly handle that case unless an absolute reference did not start with the org.

Reply:
Was wrong. We restricted it to a relative workspace in the same org. Have added OpenAPI validation for that.